### ISSUE TITLE
hidden locus collides with numbering function!

### ISSUE THREAD

**Issue by urapadmin** [urapadmin/kiosk#420]
because: if the numbering method is free input, will the user be asked?

**Comment by urapadmin** [comment]
features and surveys now obey to the default settings for numbering. see #421 and #424. The setting "numbering_method_locus" is used for the invisible locus connected to features and survey units, too. However, if the additional settings "numbering_method_feature_hidden_locus" and ""numbering_method_feature_hidden_locus" are set, those have priority. 
For features: The appropriate numbering method for the hidden locus is used both if the feature has been created and entered from the start page or if it has been created via the add feature button in survey units -> feature tab. There should be no difference in behaviour.

**Comment by urapadmin** [comment]
### @test
A hodgepodge. Most of it is already tested by #424. Except:

- [ ]  if a "free input" numbering method is selected for the creation of a hidden locus (either in survey or feature unit) that should lead to a specific error message when creating (or entering) a feature / survey unit. Always keep in mind that feature units can be added on the start page but also from within a survey! Always test that the behaviour is the same.

**Comment by lbestock** [comment]
The error message is not produced when I create the unit - I am allowed to do that. 

If I create the feature with "free input" for hidden locus from the home page, when I try to enter that unit (I can select it from the dropdown menu, but not enter it) I get the "No identification method specified: see admin" warning message.

If I create the feature with "free input" for hidden locus from within a survey unit, when I try to enter that unit (which does get created again, so I can see it in the list of features in the survey unit) I get a different error message: "Error in CreateForeinRecord: is not a valid value".

So basically, it works in that it keeps those from being functional features. But the error messages are different and only the first would tell a user anything of use to help overcome it. And I believe you wanted to stop it at the creation, not the entering, point?

**Comment by lbestock** [comment]
Oh it's a two-step error message with the feature created in a survey unit. The next one is your handmade one - "Error, for some reason it was impossible to create a unit, please talk to your admin!" But the unit still appears in the list.

**Comment by urapadmin** [comment]
![image](https://user-images.githubusercontent.com/38838314/75949310-cfdf7680-5e73-11ea-876a-e4855735e26f.png)

**Comment by lbestock** [comment]
v. 8.9.11 Alice USTP, same problems as before. Can create the feature units with the "free input for invisible locus" identification method, but then cannot enter them. The first screen shot here is the error message I got on trying to enter the feature I created from the home page:
![image](https://user-images.githubusercontent.com/49291909/79074155-c9d18680-7cb8-11ea-9543-5af00b206053.png)
Then this is what I see if I try to enter the feature I have created from a survey unit:
![image](https://user-images.githubusercontent.com/49291909/79074164-d9e96600-7cb8-11ea-8b80-d7c2348c92a7.png)

**Comment by urapadmin** [comment]
What should happen is that when entering a feature that is supposed to create the hidden locus with idgen_input the first error message appears and keeps you from entering the feature recording. Not matter from where you try (either from the start page or from the survey unit). It should say now "Wrong identification method" in the title of the error message and give you the advice to change the identification method for hidden loci.

**Comment by lbestock** [comment]
8.9.15
Works when trying to enter the feature from home. Can't yet test if it works when entering the feature from a survey unit because there's a bug in creating survey units. Test from survey unit once #550 is taken care of.

**Comment by lbestock** [comment]
8.9.16
Works. But creates a new bug, see #553.
